packages/astro/src/core/build/plugins/plugin-actions.ts

```diff
@@ -1,20 +1,17 @@
+import type { Plugin as VitePlugin } from 'vite';
 import { vitePluginActionsBuild } from '../../../actions/vite-plugin-actions.js';
 import type { BuildInternals } from '../internal.js';
-import type { AstroBuildPlugin } from '../plugin.js';
 import type { StaticBuildOptions } from '../types.js';
 
 export function pluginActions(
 	opts: StaticBuildOptions,
 	internals: BuildInternals,
-): AstroBuildPlugin {
+): VitePlugin {
+	const plugin = vitePluginActionsBuild(opts, internals);
 	return {
-		targets: ['server'],
-		hooks: {
-			'build:before': () => {
-				return {
-					vitePlugin: vitePluginActionsBuild(opts, internals),
-				};
-			},
+		...plugin,
+		applyToEnvironment(environment) {
```

Comment:
Is there a reason why `applyToEnvironment` isn't directly part of `vitePluginActionsBuild`? Probably relevant for other plugins too

Reply:
Yeah we can probably get rid of this special build plugin all together as all it does is add applyToEnvironment to the existing plugin.